test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatConfigTests.cs

```diff
@@ -407,6 +519,46 @@ public void ConfigGetCommand_NullGetLogger_Fail()
             Assert.Throws<ArgumentNullException>(() => ConfigGetRunner.Run(args, null));
         }
 
+        [Fact]
+        public void ConfigSetCommand_WithInvalidConfigKey_ThrowsCommandException()
+        {
+            // Arrange & Act
+            using var testInfo = new TestInfo("NuGet.Config");
+            var key = "InvalidConfigKey123";
+            var value = "https://TestRepo2/ES/api/v2/package";
+            var filePath = Path.Combine(testInfo.WorkingPath, "NuGet.Config");
+
+            var result = CommandRunner.Run(
+                DotnetCli,
+                Directory.GetCurrentDirectory(),
+                $"{XplatDll} config set {key} {value} --configfile {filePath}",
+                waitForExit: true);
+            var expectedError = string.Format(CultureInfo.CurrentCulture, Strings.Error_ConfigSetInvalidKey, key);
+
+            // Assert
+            DotnetCliUtil.VerifyResultFailure(result, expectedError);
+        }
+
+        [Fact]
+        public void ConfigSetCommand_WithNullArgs_ThrowsArgumentNullException()
```

Comment:
By its nature, this is a unit test since it's just testing the unit (the Run method) in isolation and does not require any other resources that are needed in a functional test. This test should be moved into the Unit Test assembly.